RevenueCat/Scripts/Purchases.cs

```diff
@@ -1271,6 +1272,64 @@ public void RedeemWebPurchase(WebPurchaseRedemption webPurchaseRedemption, Redee
         _wrapper.RedeemWebPurchase(webPurchaseRedemption);
     }
 
+    public delegate void GetEligibleWinBackOffersForProductFunc(WinBackOffer[] winBackOffers, Error error);
+
+    private GetEligibleWinBackOffersForProductFunc GetEligibleWinBackOffersForProductCallback { get; set; }
+
+    /// <summary>
+    /// Gets eligible win-back offers for a given store product. Only available on iOS 18.0+ with StoreKit 2.
+    /// Returns an error if the platform is not iOS 18.0+ or if StoreKit 2 is not used.
+    /// </summary>
+    /// <param name="storeProduct">The Purchases.StoreProduct to get win-back offers for</param>
+    /// <param name="callback">A callback receiving an array of Purchases.WinBackOffer objects or an error if unsuccessful</param>
+    public void GetEligibleWinBackOffersForProduct(StoreProduct storeProduct, GetEligibleWinBackOffersForProductFunc callback)
+    {
+        GetEligibleWinBackOffersForProductCallback = callback;
+        _wrapper.GetEligibleWinBackOffersForProduct(storeProduct);
+    }
+
+    public delegate void GetEligibleWinBackOffersForPackageFunc(WinBackOffer[] winBackOffers, Error error);
```

Comment:
We could potentially have a single delegate for both methods in the API? Considering they return the same data.

Reply:
Yeah, we probably could! I only kept them separate just in case they have to return different things in the future 🤷‍♂️